### PR TITLE
update primary

### PR DESCRIPTION
**Pull request by dbjsdev**
This is the update isPrimary functionality. 

Updates the user to primary and sets any other users who are currently primary to not primary

### CODE REVIEW

File: server/routes/accounts/user.js [hunk 1]
```diff
@@ -146,8 +146,32 @@ router.route('/establishment/:id/:userId').put(async (req, res) => {
 
             // this is an update to an existing User, so no mandatory properties!
             if (isValidUser) {
-                await thisUser.save(req.username, expiresTTLms);
 
+                await models.sequelize.transaction(async t => {
+
+                    if(thisUser._isPrimary){
```
Comment (wozzer72) [inline server/routes/accounts/user.js]: This should be encapsulated within the User entity - and not within the endpoint

File: server/models/classes/user.js [hunk 2]
```diff
@@ -427,19 +427,42 @@ class User {
...
                 // need to update the existing User record and add an
                 //  updated audit event within a single transaction
                 await models.sequelize.transaction(async t => {
+
                     // the saving of an User can be initiated within
                     //  an external transaction
                     const thisTransaction = externalTransaction ? externalTransaction : t;
 
+                    if(this._isPrimary){
+                        console.log('inere')
```
Comment (ChazUK) [inline server/models/classes/user.js]: Remove console.log

File: server/routes/accounts/user.js [hunk 3]
```diff
@@ -146,8 +146,11 @@ router.route('/establishment/:id/:userId').put(async (req, res) => {
 
             // this is an update to an existing User, so no mandatory properties!
             if (isValidUser) {
-                await thisUser.save(req.username, expiresTTLms);
 
+                await models.sequelize.transaction(async t => {
+                    await thisUser.save(req.username, expiresTTLms, t);
+                });
```
Comment (wozzer72) [inline server/routes/accounts/user.js]: Why this is being wrapped in an external transaction. The `User::save` encapsulates it's own transaction. Just restore the code.

File: server/models/classes/user.js (hunk 2 above)
Comment (dbjsdev) [reply server/models/classes/user.js]: Stay comment removed

File: server/routes/accounts/user.js (hunk 3 above)
Comment (dbjsdev) [reply server/routes/accounts/user.js]: Sequelize transaction removed

**Pull request merged**